src/main/java/com/synopsys/integration/alert/channel/jira/common/JiraTestIssueRequestCreator.java

```diff
@@ -95,6 +93,18 @@ public IssueTrackerRequest createRequest(IssueOperation operation, String messag
         return null;
     }
 
+    private IssueResolutionRequest createResolveIssueRequest(String providerName, LinkableItem topicItem, LinkableItem subTopicItem, Set<ComponentItem> componentItems, ComponentItem arbitraryItem,
+        IssueSearchProperties issueSearchProperties) {
+        IssueContentModel contentModel = jiraMessageParser.createIssueContentModel(providerName, IssueResolutionRequest.OPERATION, topicItem, subTopicItem, componentItems, arbitraryItem);
+        return IssueResolutionRequest.of(issueSearchProperties, contentModel);
+    }
+
+    private IssueCreationRequest createCreateOrUpdateIssueRequest(String providerName, LinkableItem topicItem, LinkableItem subTopicItem, Set<ComponentItem> componentItems, ComponentItem arbitraryItem,
+        IssueSearchProperties issueSearchProperties) {
+        IssueContentModel contentModel = jiraMessageParser.createIssueContentModel(providerName, IssueCreationRequest.OPERATION, topicItem, subTopicItem, componentItems, arbitraryItem);
+        return IssueCreationRequest.of(issueSearchProperties, contentModel);
+    }
```

Comment:
I think we could take this opportunity to use a bifunction here. The only difference is the Class calling of.